standalone-metastore/metastore-server/src/test/java/org/apache/hadoop/hive/metastore/InjectableBehaviourObjectStore.java

```diff
@@ -156,6 +167,19 @@ public NotificationEventResponse getNextNotification(NotificationEventRequest rq
     return getNextNotificationModifier.apply(super.getNextNotification(rqst));
   }
 
+  @Override
+  public void addNotificationEvent(NotificationEvent entry) throws MetaException {
+    if (addNotificationEventModifier != null) {
+      Boolean success = addNotificationEventModifier.apply(entry);
+      if ((success != null) && !success) {
+        throw new MetaException("InjectableBehaviourObjectStore: Invalid addNotificationEvent operation on DB: "
+                + entry.getDbName() + " table: " + entry.getTableName() + " event : " + entry.getEventType());
+      }
```

Comment:
Need to call super.addNotificationEvent(entry); if modifier method returns true.

Reply:
done